python/ray/data/impl/arrow_block.py

```diff
@@ -42,7 +42,17 @@ def items(self) -> Iterator[Tuple[str, Any]]:
         return self.as_pydict().items()
 
     def __getitem__(self, key: str) -> Any:
-        return self._row[key][0].as_py()
+        col = self._row[key]
+        if len(col) == 0:
+            return None
+        else:
+            item = col[0]
+            if isinstance(item, np.ndarray):
+                # Tensor columns have ndarrays as row values.
+                return item
+            else:
+                # Assuming that this is a pyarrow.Scalar value.
+                return item.as_py()
```

Comment:
Slightly more general approach might be to only convert if isinstance(pyarrow.Scalar) or hasattr(item, "as_py").